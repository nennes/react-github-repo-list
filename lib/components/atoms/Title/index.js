/* @flow */

import React from "react";

const Title = ({ title }: { title: string }) => {
  return (
    <span className="repo__title">
      <h3>
        {title}
      </h3>
    </span>
  );
};

export default Title;
