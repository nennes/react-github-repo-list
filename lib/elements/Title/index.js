import React from "react";
import PropTypes from "prop-types";

const Title = ({ title }) => {
  return (
    <span className="repo__title">
      <h3>
        {title}
      </h3>
    </span>
  );
};

Title.propTypes = {
  title: PropTypes.string.isRequired
};

export default Title;
