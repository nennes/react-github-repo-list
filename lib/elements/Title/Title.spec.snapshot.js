/* global test, expect */

import React from "react";
import Title from "./index";
import renderer from "react-test-renderer";

test("Title element renders correctly", () => {
  const element = renderer.create(
    <Title title="super-interesting-repo-title" />
  );
  expect(element.toJSON()).toMatchSnapshot();
});
