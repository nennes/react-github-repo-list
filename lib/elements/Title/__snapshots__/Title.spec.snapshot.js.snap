// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title element renders correctly 1`] = `
<span
  className="repo__title"
>
  <h3>
    Super cool library for React
  </h3>
</span>
`;
